File: takeItEasy-domain/src/main/java/com/sudosoo/takeiteasy/dto/post/PostDetailResponseDto.java
package com.sudosoo.takeiteasy.dto.post;

import com.sudosoo.takeiteasy.aspect.notice.NotifyInfo;
import com.sudosoo.takeiteasy.dto.comment.CommentResponseDto;
import lombok.Getter;
import org.springframework.data.domain.Page;

@Getter
public class PostDetailResponseDto implements NotifyInfo {

    private Long postId;
    private String title;
    private String content;
    private String writerName;
    private Page<CommentResponseDto> commentsResponseDto;

    public PostDetailResponseDto(Long postId, String title, String content,
                                 String writerName, Page<CommentResponseDto> comments) {
        this.postId = postId;
        this.title = title;
        this.content = content;
        this.writerName = writerName;
        this.commentsResponseDto = comments;
    }

    @Override
    public String getReceiverName() {
        return this.writerName;
    }

    @Override
    public String getNotifyContent() {
        return this.content;
    }
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/application/dto/post/PostDetailResponseDto.kt
package com.sudosoo.takeItEasy.application.dto.post

import com.sudosoo.takeItEasy.application.dto.comment.CommentResponseDto
import com.sudosoo.takeItEasy.domain.entity.Post
import org.springframework.data.domain.Page

class PostDetailResponseDto(
    val postId: Long, val writerName: String,
    val title: String, val content: String, val comments: MutableList<CommentResponseDto>
){

    constructor(post : Post, comments: MutableList<CommentResponseDto>): this (
            postId = post.id,
            writerName = post.writerName,
            title = post.title,
            content = post.content ,
            comments = comments
        )

}